### PR TITLE
Handle completely skipped message sends.

### PR DESCRIPTION
**Pull request by charlesmchen-signal**
PTAL @michaelkirk

### CODE REVIEW

File: SignalServiceKit/src/Messages/OWSMessageSender.m [hunk 1]
```diff
@@ -501,6 +501,12 @@ - (void)sendMessageToService:(TSOutgoingMessage *)message
                 }];
             }
 
+            if (sendingRecipientIds.count < 1) {
+                // All recipients are already sent or can be skipped.
+                successHandler();
+                return;
```
Comment (charlesmchen-signal) [inline SignalServiceKit/src/Messages/OWSMessageSender.m]: For example, if you create a group with no members, there's no need to send a message and this is not an error.

Comment (michaelkirk-signal) [reply SignalServiceKit/src/Messages/OWSMessageSender.m]: Hmm... so what's the difference in behavior?

Originally:

- Create a group with no members
- Send
- Shows as failed

And now:

- Create a group with no members
- Send
- Shows as sent (but not delivered)

Is that right?

Comment (charlesmchen-signal) [reply SignalServiceKit/src/Messages/OWSMessageSender.m]: It wasn't message sends to empty group that was failing, it was the "group update/creation" message itself that was failing.

Comment (charlesmchen-signal) [reply SignalServiceKit/src/Messages/OWSMessageSender.m]: But it's worth noting that this will also affect a number of other edge cases.

* You're sending to a group with 1 member and you learn that that member has left the group between the time a message is enqueued and sent.
* You're sending to a group with 1 member and that member is blocked.

Comment (charlesmchen-signal) [reply SignalServiceKit/src/Messages/OWSMessageSender.m]: I'll do my best to write a test plan that covers these and any other edge cases I can think of.

Comment (michaelkirk-signal) [reply SignalServiceKit/src/Messages/OWSMessageSender.m]: I'm just trying to understand the new behavior, from the users perspective. 

What behavior does this change (in terms of what the user sees now?) And what did they see before this change?

Comment (michaelkirk-signal) [reply SignalServiceKit/src/Messages/OWSMessageSender.m]: I guess you can cover that in the test plan - code changes LGTM

Comment (michaelkirk-signal) [reply SignalServiceKit/src/Messages/OWSMessageSender.m]: Thanks for the clarifications!
